### ISSUE TITLE
Internal crash when another TestCoroutineScope is passed to test()

### ISSUE THREAD

**Issue by Syex** [ologe/flow-test-observer#3]
Hey,

thanks for your library, really helpful.

I just had an issue that the lib apparently crashed internally without any error log and test still passes.
Look at this simple test:

```
    private val testCoroutineScope = TestCoroutineScope()

    @Test
    fun `exemplary test`() = runBlockingTest {
        val flow = flow {
            emit(1)
            emit(2)
        }

        flow.test(testCoroutineScope) {
            fail("")
        }
    }
```

This test passes. The mistake here is that I passed `testCoroutineScope` to the `test()` method. Using `this` from `runBlockingTest` scope makes the test fail as expected. I tracked down the problem to `BaseTestFlowCollector.setup()`, but sadly cannot step into the method while debugging.

Maybe you can look into this issue and try to at least crash with some message, as this was really tricky to find.

**Comment by ologe** [comment]
Hi @Syex, thank you for the report! I'll take a look as soon as I can. 
In the meanwhile, you can try to use a junit rule, something like [this](https://github.com/ologe/canaree-music-player/blob/master/test-shared/src/main/java/dev/olog/test/shared/MainCoroutineRule.kt) will work.

**Comment by ologe** [comment]
Just published 1.4.1 with the fix :)

It seems that exceptions were swallowed inside the coroutine launched by `suspend fun <T> Flow<T>.test`.
